### PR TITLE
changelog: add v0.4.3

### PR DESCRIPTION
**Pull request by lgierth**
Needs a bit more wording for the "notable changes".

License: MIT
Signed-off-by: Lars Gierth <larsg@systemli.org>

### CODE REVIEW

**Comment by lgierth** [comment]
Travis CI failure is this one: https://github.com/ipfs/go-ipfs/issues/2978

**Comment by lgierth** [comment]
This is more or less complete now -- only the stdin rework is todo.

**Comment by lgierth** [comment]
- Capitalized @Kubuxu's handle
- Removed OpenBSD entry -- it's not actually working yet

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,126 @@
...
+  - The internal naming of blocks in the blockstore has changed, which requires a migration of the fs-repo, from version 3 to 4. (@whyrusleeping, [ipfs/go-ipfs#2903](https://github.com/ipfs/go-ipfs/pull/2903))
+  - We now automatically raise the file descriptor limit to 1024 if neccessary. (@whyrusleeping, [ipfs/go-ipfs#2884](https://github.com/ipfs/go-ipfs/pull/2884), [ipfs/go-ipfs#2891](https://github.com/ipfs/go-ipfs/pull/2891))
+  - After a long struggle with deadlocks and hanging connections, we've decided to disable the uTP transport by default for now. (@whyrusleeping, [ipfs/go-ipfs#2840](https://github.com/ipfs/go-ipfs/pull/2840), [ipfs/go-libp2p-transport@88244000](https://github.com/ipfs/go-libp2p-transport/commit/88244000f0ce8851ffcfbac746ebc0794b71d2a4))
+  - There is now documentation for the configuration options in `docs/config.md`. (@whyrusleeping, [ipfs/go-ipfs#2974](https://github.com/ipfs/go-ipfs/pull/2974))
+  - TODO: stdin handling
```
Comment (whyrusleeping) [inline CHANGELOG.md]: TODO

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,5 +1,126 @@
...
 
+### 0.4.3-rc1 - 2016-07-23
+
+This is a maintenance release which comes with a couple of nice enhancements, and improves the performance of Storage, Bitswap, as well as Content and Peer Routing. It also introduces a handful of new commands and options, and fixes a good bunch of bugs.
+
+This is the first Release Candidate. Unless there are vulnerabilities or regressions discovered, the final 0.4.3 release will happen about one week from now.
+
+- Security Vulnerability
+
+  - We are thankful that a group of independent researchers made us aware of a vulnerability which the `master` branch suffered from for about 3 weeks. It allowed an attacker to e.g. use an iframe to request malicious HTML and JS from the API of a local go-ipfs node, and use this to then extract the node's private key. We fixed this issue by reintroducing restrictions on which particular objects can be loaded through the API (@lgierth, [ipfs/go-ipfs#2949](https://github.com/ipfs/go-ipfs/pull/2949)), and by completely excluding the private key from the API (@Kubuxu, [ipfs/go-ipfs#2957](https://github.com/ipfs/go-ipfs/pull/2957)). We will also work on more hardening of the API in the next release.
```
Comment (RichardLitt) [inline CHANGELOG.md]: No need for e.g.

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,5 +1,126 @@
...
+  - Fix progress bars in `ipfs add` and `ipfs get`. (@whyrusleeping, [ipfs/go-ipfs#2893](https://github.com/ipfs/go-ipfs/pull/2893), [ipfs/go-ipfs#2948](https://github.com/ipfs/go-ipfs/pull/2948))
+  - Make sure files added through `ipfs files` are pinned and don't get GC'd. (@kevina, [ipfs/go-ipfs#2872](https://github.com/ipfs/go-ipfs/pull/2872))
+  - Fix copying into directory using `ipfs files cp`. (@whyrusleeping, [ipfs/go-ipfs#2977](https://github.com/ipfs/go-ipfs/pull/2977))
+  - Fix `ipfs version --commit` with Docker containers. (@lgierth, [ipfs/go-ipfs#2734](https://github.com/ipfs/go-ipfs/pull/2734))
+  - Run `ipfs diag` commands in the daemon instead of the CLI. (@Kubuxu, [ipfs/go-ipfs#2761](https://github.com/ipfs/go-ipfs/pull/2761))
+  - Fix protobuf encoding on the API and in commands. (@stebalien, [ipfs/go-ipfs#2516](https://github.com/ipfs/go-ipfs/pull/2516))
+  - Fix goroutine leak in `/ipfs/ping` protocol handler. (@whyrusleeping, [ipfs/go-libp2p#58](https://github.com/ipfs/go-libp2p/pull/58))
+  - Fix `--flags` for all commands. (@Kubuxu, [ipfs/go-ipfs#2773](https://github.com/ipfs/go-ipfs/pull/2773))
```
Comment (RichardLitt) [inline CHANGELOG.md]: How?

File: CHANGELOG.md [hunk 4]
```diff
@@ -1,5 +1,126 @@
...
+  - Fix go-ipfs/js-ipfs interoperability issues in SPDY. (@whyrusleeping, [whyrusleeping/go-smux-spdystream@fae17783](https://github.com/whyrusleeping/go-smux-spdystream/commit/fae1778302a9e029bb308cf71cf33f857f2d89e8))
+  - Fix a logging race condition during shutdown. (@Kubuxu, [ipfs/go-log#3](https://github.com/ipfs/go-log/pull/3))
+  - Prevent DHT connection hangs. (@whyrusleeping, [ipfs/go-ipfs#2826](https://github.com/ipfs/go-ipfs/pull/2826), [ipfs/go-ipfs#2863](https://github.com/ipfs/go-ipfs/pull/2863))
+  - Fix NDJSON output of `ipfs refs local`. (@Kubuxu, [ipfs/go-ipfs#2812](https://github.com/ipfs/go-ipfs/pull/2812))
+  - Fix race condition in NAT detection. (@whyrusleeping, [ipfs/go-libp2p#69](https://github.com/ipfs/go-libp2p/pull/69))
+  - Fix error messages. (@whyrusleeping, @Kubuxu, [ipfs/go-ipfs#2905](https://github.com/ipfs/go-ipfs/pull/2905), [ipfs/go-ipfs#2928](https://github.com/ipfs/go-ipfs/pull/2928))
+
+- Enhancements
+
+  - Increase maximum object size to 2 MiB. (@kpcyrd, [ipfs/go-ipfs#2980](https://github.com/ipfs/go-ipfs/pull/2980))
```
Comment (RichardLitt) [inline CHANGELOG.md]: Might be good to have 'why' for all of these. But I guess they can follow up.

File: CHANGELOG.md [hunk 5]
```diff
@@ -1,5 +1,126 @@
...
+  - Allow setting position of default values in command option descriptions. (@Kubuxu, [ipfs/go-ipfs#2744](https://github.com/ipfs/go-ipfs/pull/2744))
+  - Set explicit default values for boolean command options. (@richardlitt, [ipfs/go-ipfs#2657](https://github.com/ipfs/go-ipfs/pull/2657))
+  - Autogenerate command synopsises. (@Kubuxu, [ipfs/go-ipfs#2785](https://github.com/ipfs/go-ipfs/pull/2785))
+  - Fix and improve lots of documentation. (@richardlitt, [ipfs/go-ipfs#2741](https://github.com/ipfs/go-ipfs/pull/2741), [ipfs/go-ipfs#2781](https://github.com/ipfs/go-ipfs/pull/2781))
+  - Improve command descriptions to fit a width of 78 characters. (@richardlitt, [ipfs/go-ipfs#2779](https://github.com/ipfs/go-ipfs/pull/2779), [ipfs/go-ipfs#2780](https://github.com/ipfs/go-ipfs/pull/2780), [ipfs/go-ipfs#2782](https://github.com/ipfs/go-ipfs/pull/2782))
+  - Fix filename conflict in the debugging guide. (@Kubuxu, [ipfs/go-ipfs#2752](https://github.com/ipfs/go-ipfs/pull/2752))
+  - Decapitalize log messages, according to Golang style guides. (@richardlitt, [ipfs/go-ipfs#2853](https://github.com/ipfs/go-ipfs/pull/2853))
```
Comment (RichardLitt) [inline CHANGELOG.md]: Uppercase: RichardLitt

File: CHANGELOG.md [hunk 6]
```diff
@@ -1,5 +1,126 @@
...
+  - Add Github Issue template. (@chriscool, [ipfs/go-ipfs#2786](https://github.com/ipfs/go-ipfs/pull/2786))
+  - Apply standard-readme to the README file. (@richardlitt, [ipfs/go-ipfs#2883](https://github.com/ipfs/go-ipfs/pull/2883))
+  - Fix issues pointed out by `govet`. (@Kubuxu, [ipfs/go-ipfs#2854](https://github.com/ipfs/go-ipfs/pull/2854))
+  - Clarify `ipfs get` error message. (@whyrusleeping, [ipfs/go-ipfs#2886](https://github.com/ipfs/go-ipfs/pull/2886))
+  - Remove dead code. (@whyrusleeping, [ipfs/go-ipfs#2819](https://github.com/ipfs/go-ipfs/pull/2819))
+  - Add changelog for v0.4.3. (@lgierth, [ipfs/go-ipfs#2984](https://github.com/ipfs/go-ipfs/pull/2984))
+
+- Tests & CI
+
+  - Fix flakey `ipfs mount` sharness test by using the `iptb` tool. (@noffle, [ipfs/go-ipfs#2707](https://github.com/ipfs/go-ipfs/pull/2707))
```
Comment (RichardLitt) [inline CHANGELOG.md]: Flaky.

**Comment by RichardLitt** [comment]
Looks pretty good to me!

File: CHANGELOG.md (hunk 2 above)
Comment (jbenet) [inline CHANGELOG.md]: - Note that this was only in master and not in a release, 
- reiterate that if people stay on the releases as recommended they shouldn't have seen it at all.
- ask for more watchful eyes on security on PRs and merges

Comment (jbenet) [inline CHANGELOG.md]: Just saw the next paragraph sorry. You already did some of that.

**Comment by whyrusleeping** [comment]
LGTM, lets do this

**Pull request merged**